keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -315,18 +315,26 @@ index efbd72c133..f86a1a5512 100644
  }
 
 +message TopologyInfo {
-+  optional int32 socketID = 1 [default = -1];
++  NUMANode node = 1;
++}
++
++message NUMANode {
++    int64 ID = 1;
```

Comment:
Please, don't use `int` here. It should be list of nodes, to match format of cgroup's `cpuset/cpuset.mems`

Reply:
This is based on the merged API change here: https://github.com/kubernetes/kubernetes/pull/74423

Reply:
It is really sad that there is practice "design follows the code".

Reply:
The original field was an int so the code has followed the design. However, there was concern on the default value (brought up by yourself on the code PR itself) and a decision was made to have a nested struct so that in the event a device plugin author does not send back Topology Information it will be an empty struct.
https://github.com/kubernetes/kubernetes/pull/74423#discussion_r289183175

Reply:
@ConnorDoyle well, if I put on hat of device plugin developer, it will be indeed better to have it as a string. It means less formats to convert between (what I get from kernel, I need to parse, but I can use it directly in API and, as you mentioned, we already have code to work with that format). With a hat of kubernetes developer, I see benefits with defining it as `repeated NUMANode`. It means for plugin developer it will need to transform `0-3` to `[]{0,1,2,3}`, but on protocol side we will have ability to validate that every single passed value is correctly typed, and not some garbage that can be in case of strings.

So, I think, option 1, what @klueska did in kubernetes/kubernetes#82020 is good. 
It is good from the point that it can be 0 or more elements, and `message TopologyInfo` can be later extended to have more similar entries like `NUMANode` for other topology attributes.

Makes sense?

@lmdaly sorry for confusion from my side, I misread patch a bit when I was adding comment on Monday, din't notice nested struct and realize that only later when I continued other thread with @klueska.